lib/condor.py

```diff
@@ -227,12 +221,93 @@ def submit_dag(
                 sys.stderr.write(
                     f"Error: Child was terminated by signal {-output.returncode}"
                 )
-            else:
-                if "outdir" in vargs:
-                    print(
-                        f"Output will be in {vargs['outdir']} after running jobsub_transfer_data."
-                    )
         except OSError as e:
             print("Execution failed: ", e)
 
     return submit(subfile, vargs, schedd_name)
+
+
+class JobIdError(Exception):
+    pass
+
+
+class Job:
+    """
+    Job represents a single HTCondor batch job or cluster with an id like
+        <cluster>[.<process>]@<schedd>
+
+    where if <process> is specified it is a single job, or if not then a cluster
+    of jobs. Examples:
+
+
+        123@schedd.example.com
+        123.0@schedd.example.com
+        123.456@schedd.example.com
+
+    """
+
+    id: str
+    seq: int
+    proc: int
+    schedd: str
+    cluster: bool
+
+    _id_regexp = re.compile(r"(\d+)(?:\.(\d+))?@([\w\.]+)")
+
+    def __init__(self, job_id: str):
+        self.id = job_id
+        m = Job._id_regexp.match(job_id)
+        if m is None:
+            raise JobIdError(f'unable to parse job id "{job_id}"')
+        try:
+            self.seq = int(m.group(1))  # seq is required
+            if m.group(2) is None:  # proc is optional
+                self.cluster = True
+                self.proc = 0
+            else:
+                self.cluster = False
+                self.proc = int(m.group(2))
+            self.schedd = m.group(3)  # schedd is required
+        except TypeError as e:
+            raise JobIdError(f'error when parsing job id "{job_id}"') from e
+
+    def __str__(self) -> str:
+        if self.cluster:
+            return f"{self.seq}@{self.schedd}"
+        return f"{self.seq}.{self.proc}@{self.schedd}"
+
+    def _get_schedd(self) -> htcondor.htcondor.Schedd:
```

Comment:
Do we anticipate using these protected methods outside of the `Job` class?  If not, this and `_constraint` should probably be protected, as in `__get_schedd` and `__constraint`

Reply:
I decided single underscore was appropriate, since they're not intended to be used outside the class, but there's no reason they couldn't, and they have no side effects. But since you brought it up, PEP8 also says

> Generally, double leading underscores should be used only to avoid name conflicts with attributes in classes designed to be subclassed.

Which `Job` isn't.

Reply:
Fair enough.